src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzerTests.GuardedCallsTests.cs

```diff
@@ -3612,6 +3612,38 @@ public static partial class ProcessExtensions
             await VerifyAnalyzerAsyncCs(source);
         }
 
+        [Fact, WorkItem(4372, "https://github.com/dotnet/roslyn-analyzers/issues/4372")]
+        public async Task LoopWithinGuardCheck_02()
+        {
+            var source = @"
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+
+namespace Repro
+{
+    public class C
+    {
+        public static void M()
+        {
+            for (var i = 0; i < 2; i++)
```

Comment:
If putting the platform check outside of the loop would make difference or could repro the bug? If so could we add or update the test with that?
```csharp
            if (OperatingSystemHelper.IsWindows())
            {
                for (var i = 0; i < 2; i++)
                {
                    WindowsSupported();
                }
            }
```

Reply:
Sure, I'll add another test for that.